### ISSUE TITLE
Show Layout

### ISSUE THREAD

**Issue by nkabbara** [ranelpadon/ngram-type#6]
Thanks for ngram-type! Do you plan on having a layout shown around the typing area? 

That would be great for learning new layouts.

**Comment by ranelpadon** [comment]
@nkabbara You mean something like this?
https://ranelpadon.github.io/keyzen-colemak-dh/

That's my other project. You could choose the layout there. And if you're still not familiar with the layout/keys placement, might be better to learn it first like using that **KeyZen Colemak DH** for some time before proceeding to **Ngram Type**, or practicing alongside with it? Actually, I'm still using **KeyZen Colemak DH** now using the **Custom** mode to improve my typing speed/accuracy in punctuations/numbers. :)

**Comment by ranelpadon** [comment]
@nkabbara Interesting, so these are the placements of ISRT?
```
y c l m k    z f u , '
i s r t g    p n e a o ;
q v w d j    b h / . x

y c l m k z f u , '
 i s r t g p n e a o ;
  q v w d j b h / . x

y c l m k z f u , '
 i s r t g p n e a o ;
  v w d j q b h / . x
```
Which of these 3 are you using though?

Do you want me to add in the **KeyZen Colemak DH** page as another layout/option? I could add it if you want. Not sure if I could add/install it in my Mac then get the layout screenshot, or maybe you could provide me a photo also.

**Comment by ranelpadon** [comment]
I could generate a photo similar to this I think (ANSI/ANSI Angle-Modded) but with ISRT keys if this will suffice, unless you're using the ortholinear keyboard:
<img width="517" alt="Screen Shot 2021-12-14 at 11 27 15 PM" src="https://user-images.githubusercontent.com/4292088/146028249-11c65c78-ecb7-4b5f-81d2-96c946c3ee42.png">

**Comment by nkabbara** [comment]
Thank you so much for looking into it!!!

This is a photo of my layout. It's lifted from Ben Vallack's vid:
https://www.youtube.com/watch?v=8wZ8FRwOzhU&t=161s

[image: layer-1.png]

Basically a 34key ortho isrt:

y c l m k    z f u , <-
i s r t g    p n e a o
q v w d j    b h / . x

**Comment by ranelpadon** [comment]
@nkabbara The 3 ISRT layouts above have this in the top row:
`y c l m k       z f u , '`

but you have 
`y c l m k       z f u , <-`

- So, you used `<-` instead of `'`? Is that allowed deviation for ISRT, or non-standard, and just personal thing? 
- Does `<-` has 2 characters or simply mean the `<`?

Actually, in the meantime, you could try that layout now, by clicking the **Custom** mode/layout:
https://ranelpadon.github.io/keyzen-colemak-dh/

then input these characters: main row -> top row -> bottom row. Then, in each row: alternate left/right fingers' keys, middle keys with last priorities. Specifically:
`tnresaiogpmfluc,y<kzdhw/v.qxjb`

Tried the ISRT now in the site using the **Custom** menu, works ok, it will introduce the keys gradually. Set the Intensity to 9 also (lower right). Refresh the site after inputting the custom layout keys and/or updating the intensity:

<img width="1440" alt="Screen Shot 2021-12-15 at 2 15 28 AM" src="https://user-images.githubusercontent.com/4292088/146056546-ed707247-caa1-41fe-ae14-3c3aef3d5b84.png">

**Comment by nkabbara** [comment]
Thank you! I'll give that a try.

> So, you used <- instead of '? Is that allowed deviation for ISRT, or non-standard, and just personal thing?
> Does <- has 2 characters or simply mean the <?

Sorry I wasn't clear here. I used '<-' to signify backspace. I replaced the "'" in standard isrt with a backspace for top right key. I also use an ortho keyboard. Kinda like below but with backspace instead of the shown '.

<img width="444" alt="Screen Shot 2021-12-14 at 12 35 51 PM" src="https://user-images.githubusercontent.com/31865/146059266-08a563b2-7309-4e39-a609-f92836987549.png">
photo from monkeytype.


Is this open source by any chance? Maybe I can clone, modify and run for my custom setup?

**Comment by ranelpadon** [comment]
Ah I see, kinda tricky then how to represent the Backspace/Delete key as character to type, since usually they are not practiced, even in other typing sites. Yep, you could fork it, or submit a PR, up to you:
> Is this open source by any chance? Maybe I can clone, modify and run for my custom setup?

Otherwise, you could just use the `Custom` layout mode, and just ignore the `Backspace`, or practice it last, or manually.

**Comment by nkabbara** [comment]
Perfect. Will try custom and report back.

So you recommendation for learning a new layout is keyzen, then ngram type,
then something like monkeytype? I'm guessing a combo of these works as well.


-Nash

**Comment by ranelpadon** [comment]
I remapped my number keys recently to a Numbers layer, so I use KeyZen also for that purpose, using `Custom` mode (mainly for special layout/specific keys). Likewise, improving my punctuations with it.

**Comment by nkabbara** [comment]
nice nice nice! I'm glad I opened a ticket here :)

Thank you.


-Nash

**Comment by ranelpadon** [comment]
No worries, I'm the maintainer of both projects anyway. Let me know if you encounter issues or need help. 
Cheers :)

**Comment by Zireael07** [comment]
Does keyzen remember the custom layout? I feel like it doesn't...

**Comment by ranelpadon** [comment]
@Zireael07 Yep, it does, I double-checked now. If you choose the `custom` layout, it will be saved in the browser's Local Storage including the specified custom chars, and it will persist even after the page reload.

https://ranelpadon.github.io/keyzen-colemak-dh/

<img width="1726" alt="Screenshot 2023-03-20 at 4 03 14 AM" src="https://user-images.githubusercontent.com/4292088/226206029-cc8430f0-cb66-41cd-8b6d-c83b1752228d.png">
